tt_metal/impl/buffers/buffer.cpp

```diff
@@ -240,16 +244,38 @@ std::shared_ptr<Buffer> Buffer::create(
         buffer->address_ = detail::AllocateBuffer(buffer.get());
 
         std::unique_lock lock(buffer->allocation_mutex_);
-        buffer->allocation_status_.store(AllocationStatus::ALLOCATED, std::memory_order::relaxed);
+        buffer->allocation_status_.store(AllocationStatus::ALLOCATED, std::memory_order::release);
```

Comment:
Does this really need to be made `release`? It seems like we're already protected by a lock.

Reply:
Yes. In `Buffer::address()` the first check is `allocation_status_.load(std::memory_order::acquire) != AllocationStatus::ALLOCATION_REQUESTED` performed without a lock. This is done this way for performance optimization reasons. The only two places who capture a lock are`Buffer::create` when allocating and `Buffer::address` when waiting on an allocation using condition variable.